Config.cmake/fmiplatform.cmake

```diff
@@ -14,27 +14,56 @@
 # fmi_platform(platform) sets platform to one of the by FMI defined platforms
 # win32, win64, linux32, linux64, darwin32 or darwin64 depending on cmake
 # generator
-function(fmi_platform platform)
+function(fmi_platform platform fmi_version)
+    if(fmi_version EQUAL 1 OR fmi_version EQUAL 2)
+        set(FMI_1_OR_2 TRUE)
+    else()
+        set(FMI_1_OR_2 FALSE)
+    endif()
     if(CMAKE_HOST_WIN32) #Set to true when the host system is Windows and on cygwin.
         if(CMAKE_SIZEOF_VOID_P EQUAL 4)
-            set(${platform} win32 PARENT_SCOPE)
-        else(CMAKE_SIZEOF_VOID_P EQUAL 4)
-            set(${platform} win64 PARENT_SCOPE)
-        endif(CMAKE_SIZEOF_VOID_P EQUAL 4)
-    elseif(CMAKE_HOST_APPLE) #Set to true when the host system is Apple OSX. 
+            if(FMI_1_OR_2)
+                set(${platform} win32 PARENT_SCOPE)
+            else()
+                set(${platform} x86-windows PARENT_SCOPE)
+            endif()
+        else()
+            if(FMI_1_OR_2)
+                set(${platform} win64 PARENT_SCOPE)
+            else()
+                set(${platform} x86_64-windows PARENT_SCOPE)
+            endif()
+        endif()
+    elseif(CMAKE_HOST_APPLE)
         if(CMAKE_SIZEOF_VOID_P EQUAL 4)
-            set(${platform} darwin32 PARENT_SCOPE)
-        else(CMAKE_SIZEOF_VOID_P EQUAL 4)
-            set(${platform} darwin64 PARENT_SCOPE)
-        endif(CMAKE_SIZEOF_VOID_P EQUAL 4)
+            if(FMI_1_OR_2)
+                set(${platform} darwin32 PARENT_SCOPE)
+            else()
+                set(${platform} x86-darwin PARENT_SCOPE)  # Not defined in spec
+            endif()
+        else()
+            if(FMI_1_OR_2)
+                set(${platform} darwin64 PARENT_SCOPE)
+            else()
+                set(${platform} x86_64-darwin PARENT_SCOPE)
+            endif()
+        endif()
     elseif(CMAKE_HOST_UNIX) #Set to true when the host system is UNIX or UNIX like (i.e. APPLE and CYGWIN). 

```

Comment:
In comment: APPLE already covered in the previous elseif?

Reply:
Yes, the comment kind of just explains when _UNIX is true, i.e. it's kind of a reminder that you must check APPLE before UNIX or it will get caught already